File: ITapply/ITapply.UI/itapply_desktop/lib/model/job_posting.dart
import 'package:itapply_desktop/model/job_posting_skill.dart';
import 'package:json_annotation/json_annotation.dart';

part 'job_posting.g.dart';
@JsonSerializable()
class JobPosting {
  final int id;
  final int employerId;
  final String employerName;
  final String title; // max 200 chars
  final String description; // max 10000 chars
  final String? requirements; // max 5000 chars, optional
  final String? benefits; // max 3000 chars, optional
  final int employmentType; // enum
  final int experienceLevel; // enum
  final int? locationId;
  final String? locationName;
  final int remote; // enum
  final int minSalary;
  final int maxSalary;
  final DateTime applicationDeadline;
  final DateTime postedDate;
  final int status; // enum
  final List<JobPostingSkill> skills;
  final int applicationCount;

  JobPosting({
    required this.id,
    required this.employerId,
    required this.employerName,
    required this.title,
    required this.description,
    this.requirements,
    this.benefits,
    required this.employmentType,
    required this.experienceLevel,
    this.locationId,
    this.locationName,
    required this.remote,
    required this.minSalary,
    required this.maxSalary,
    required this.applicationDeadline,
    required this.postedDate,
    required this.status,
    required this.skills,
    required this.applicationCount,
  });

  factory JobPosting.fromJson(Map<String, dynamic> json) => _$JobPostingFromJson(json);
}

File: ITapply/ITapply.Services/Database/JobPosting.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Services.Database
{
    public class JobPosting
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int EmployerId { get; set; }
        [ForeignKey("EmployerId")]
        public Employer Employer { get; set; }

        [Required]
        [StringLength(200)]
        public string Title { get; set; } // e.g., "Senior .NET Developer", "UX/UI Designer"

        [Required]
        [StringLength(10000)]
        public string Description { get; set; } // Detailed job description

        [StringLength(5000)]
        public string? Requirements { get; set; }

        [StringLength(3000)]
        public string? Benefits { get; set; }

        [Required]
        public EmploymentType EmploymentType { get; set; }

        [Required]
        public ExperienceLevel ExperienceLevel { get; set; }

        public int? LocationId { get; set; }
        [ForeignKey("LocationId")]
        public Location Location { get; set; }

        [Required]
        public Remote Remote { get; set; }

        [Range(0, int.MaxValue)]
        public int? MinSalary { get; set; }

        [Range(0, int.MaxValue)]
        public int? MaxSalary { get; set; }

        [Required]
        public DateTime ApplicationDeadline { get; set; }

        [Required]
        public DateTime PostedDate { get; set; } = DateTime.Now;

        [Required]
        public JobPostingStatus Status { get; set; } = JobPostingStatus.Active; // Default status

        // Navigation properties
        public ICollection<JobPostingSkill> JobPostingSkills { get; set; } // Required skills for this job
        public ICollection<Application> Applications { get; set; } // Applications received for this job
    }
}
